Map and waterflow unittests

sziki12 opened the pull request:
Map Creation, Waterflow and Point Calculation unit tests created.

sziki12 committed:
Test Base Defined

Created common init for tests and Map and WaterFlow test skeleton, progress in implementation

sziki12 committed:
Water flow test extension

sziki12 committed:
Tests Completed

Waterflow, Point Calculation and Map Setup tests

sziki12 committed:
Pipe Bug Fix

Pipe was leaking water without water in it.

sziki12 committed:
GUI for Unit Test

Abri02 reviewed (approved):
I've reviewed the unit tests in this pull request and found them to be thorough and well-written.

sziki12 committed:
Function Call Test and Pump fix

Waterflow function test by invocation count for different elements of the map.
Fix for Pump step, now it correctly calls the step of the neighbour pipes.
